noobaa/noobaa-core
# NS Buckets Triggers & UI Triggers Additions

### Explain the changes
#### 1. New Features / Capabilities (Sync with Liran):
- NS Buckets Triggers
- Read trigger implementation
- Triggers Tab in Function Page
- Functions Audit Log

#### 2. Existing Behavior Changes (Sync with Liran):
- Changes to triggers unit test - move to async and run on NS buckets as well

### Issues
- Fixes #5303

### Testing Instructions:
- Verify triggers on NS buckets
- Verify triggers on READ event

### Reminders:
- Create a new test - the first is the hardest
- User Experience is top priority
- Design for failure
- Keep it simple
- `npm test`
- Imagine a support call - Add diagnostics info, phonehome info, activity log events, external syslog
- Upgrade - DB schema, server platform, agents install, npm packages
